Tempo 3.0 migration: docker-compose example + doc tweaks (#7122)
grafana/tempo: docs/sources/tempo/set-up-for-tracing/setup-tempo/migrate-to-3.md

1. Verify that all components start successfully and connect to Kafka. Check that `tempo_live_store_ready` equals `1`. The log message `"live-store ready to serve queries"` confirms the live-store is ready. At this point, no traffic is flowing through the 3.0 distributors yet, so write metrics will be at zero.

   {{< admonition type="note" >}}
   On a fresh deployment with no traffic flowing yet, the live-store can take up to `live_store.readiness_max_wait` (default 30 minutes) to become ready, because there's no Kafka high-water mark to compare against. This is normal. Once traffic starts flowing in the next step, restarts are near-instant.
   {{< /admonition >}}

> 1. It's highly recommended to validate the deployment end-to-end using [Tempo Vulture](/docs/tempo/<TEMPO_VERSION>/operations/tempo-vulture/), Tempo's built-in testing tool. Vulture writes traces to the distributor and reads them back through the query frontend, validating the full write and read path. Point it at the 3.0 deployment and let it run for 10–15 minutes. Confirm that `tempo_vulture_trace_error_total` stays at zero and `tempo_vulture_trace_total` is increasing.

Reviewer: Because this is a step, we should lead with a "do this" statement. What do you think about this? Instead of saying "Highly recommended", let's just have them do it. 
```suggestion
1. Validate the deployment end-to-end using [Tempo Vulture](/docs/tempo/<TEMPO_VERSION>/operations/tempo-vulture/), Tempo's built-in testing tool. Vulture writes traces to the distributor and reads them back through the query frontend, validating the full write and read path. Point it at the 3.0 deployment and let it run for 10–15 minutes. Confirm that `tempo_vulture_trace_error_total` stays at zero and `tempo_vulture_trace_total` is increasing.
```

   Use a trace ID that you know exists in your 2.x deployment. If the query returns trace data, the new deployment is reading from storage correctly.

Before proceeding to the cutover, confirm all of the following:

- `tempo_live_store_ready` equals `1` on every live-store replica.
- `rate(tempo_block_builder_fetch_errors_total[5m])` is at or near zero on every block-builder.
- Vulture has been running for 10–15 minutes with `tempo_vulture_trace_error_total` at zero and `tempo_vulture_trace_total` increasing.
- At least one trace-by-ID query for a trace that exists in your 2.x deployment returns data when sent to the 3.0 query frontend.

> If any of these aren't satisfied, resolve them before continuing. Switching traffic with an unhealthy 3.0 deployment risks data loss for newly ingested traces.

Reviewer: ```suggestion
Resolve any issues so all of these conditions are satisfied. Switching traffic with an unhealthy 3.0 deployment risks data loss for newly ingested traces.
```
